File: lib/to_mdx/parsers/open_document/meta/keyword.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Meta
      class Keyword < Base
        attr_reader :keyword_text

        def initialize(element = nil, context = {})
          super(element, context)
          @keyword_text = element&.text&.strip

          unless @keyword_text&.empty?
            context["keywords"] ||= []
            context["keywords"] << @keyword_text
          end
        end

        def to_mdx
          ""
        end

        def to_s
          keyword_text || ""
        end

        register_for "meta:keyword"
      end
    end
  end
end

File: src/parsers/open-document/meta/keyword.ts
import type { ParseContext } from "../../base_element.ts";
import { BaseElement } from "../../base_element.ts";

export class Keyword extends BaseElement {
  private keywordText: string;

  constructor(element: Element, context: ParseContext, parent: BaseElement | null) {
    super(element, context, parent);
    this.keywordText = (element.textContent || "").trim();

    if (this.keywordText) {
      context.keywords ??= [];
      (context.keywords as string[]).push(this.keywordText);
    }
  }

  toMdx(): string {
    return "";
  }

  toString(): string {
    return this.keywordText || "";
  }

  static {
    BaseElement.registerFor.call(this, "meta:keyword");
  }
}
